react/features/app/actions.web.js

```diff
@@ -0,0 +1,173 @@
+export function appNavigate(urlOrRoom) {
+    return (dispatch, getState) => {
+        const oldDomain = getDomain(getState());
+
+        const { domain, room } = _getRoomAndDomainFromUrlString(urlOrRoom);
+
+        // TODO Kostiantyn Tsaregradskyi: We should probably detect if user is
+        // currently in a conference and ask her if she wants to close the
+        // current conference and start a new one with the new room name or
+        // domain.
+
+        if (typeof domain === 'undefined' || oldDomain === domain) {
+            // If both domain and room vars became undefined, that means we're
+            // actually dealing with just room name and not with URL.
+            dispatch(
+                _setRoomAndNavigate(
+                    typeof room === 'undefined' && typeof domain === 'undefined'
+                        ? urlOrRoom
+                        : room));
+        } else if (oldDomain !== domain) {
+            // Update domain without waiting for config to be loaded to prevent
+            // race conditions when we will start to load config multiple times.
+            dispatch(setDomain(domain));
+
+            // If domain has changed, we need to load the config of the new
+            // domain and set it, and only after that we can navigate to
+            // different route.
+            loadConfig(`https://${domain}`)
+                .then(
+                    config => configLoaded(/* err */ undefined, config),
+                    err => configLoaded(err, /* config */ undefined));
+        }
+
+        /**
+         * Notifies that an attempt to load the config(uration) of domain has
+         * completed.
+         *
+         * @param {string|undefined} err - If the loading has failed, the error
+         * detailing the cause of the failure.
+         * @param {Object|undefined} config - If the loading has succeeded, the
+         * loaded config(uration).
+         * @returns {void}
+         */
+        function configLoaded(err, config) {
+            if (err) {
+                // XXX The failure could be, for example, because of a
+                // certificate-related error. In which case the connection will
+                // fail later in Strophe anyway even if we use the default
+                // config here.
+
+                // The function loadConfig will log the err.
+                return;
+            }
+
+            // We set room name only here to prevent race conditions on app
+            // start to not make app re-render conference page for two times.
+            dispatch(setRoom(room));
+            dispatch(setConfig(config));
+            _navigate(getState());
+        }
+    };
+}
+
+/**
+ * Temporary solution. Should dispatch actions related to
+ * initial settings of the app like setting log levels,
+ * reading the config parameters from query string etc.
+ *
+ * @returns {Function}
+ */
+export function appInit() {
+    return () => {
+        init();
+    };
+}
```

Comment:
The file react/features/app/actions.web.js is practically a copy of 158 lines from react/features/app/actions.native.js with the addition of a simple function. As per the coding style, this is unacceptable.